src/pybind/mgr/cephadm/module.py

```diff
@@ -152,6 +152,21 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
     return inner
 
 
+def service_exists(service_name_position: int = 1) -> Callable:
+    """Check that a hostname exists in the inventory"""
+    def inner(func: Callable) -> Callable:
+        @wraps(func)
+        def wrapper(*args: Any, **kwargs: Any) -> Any:
+            this = args[0]  # self object
+            service_name = args[service_name_position]
+            if service_name not in [service[0] for service in this.spec_store.all_specs.items()]:
+                raise OrchestratorError(
+                    f"Cannot find service '{service_name}' in the inventory")
+            return func(*args, **kwargs)
+        return wrapper
+    return inner
```

Comment:
this function probably works, but I'm wondering if it would be simpler to just check `if service_name not in self.spec_store` inside of the `set_osd_spec` function. The `SpecStore` class has a defined `__contains__` function specifically for doing this type of check

Reply:
Addressed. Thanks for pointing this!